geps/gep-2649/index.md

```diff
@@ -0,0 +1,744 @@
+  Policies attached _higher_ up the hierarchy beat the same type of Policy
+  attached further down the hierarchy.
+* `defaults` settings operate in a "more specific beats less specific" fashion -
+  Policies attached _lower down_ the hierarchy beat the same type of Policy
+  attached further _up_ the hierarchy.
+* For `defaults`, the _most specific_ value is the one _inside the object_ that
+  the Policy applies to; that is, if a Policy specifies a `default`, and an object
+  specifies a value, the _object's_ value wins.
+* Policies interact with the fields they are controlling in a "replace value"
+  fashion.
+    * For fields where the `value` is a scalar, (like a string or a number)
+      MUST have their value _replaced_ by the value in the Policy if it wins.
+      Notably, this means that a `default` will only ever replace an empty or unset
+      value in an object. Note also that, as in Go, the empty string value is _only_
+      a string of length 0, generally represented as `""`.
+    * For fields where the value is an object, the Policy should include the fields
+      in the object in its definition, so that the replacement can be performed
+      on each field inside the object as a simple field rather than the object as
+      a whole.
+    * For fields where the final value is non-scalar, but is not an _object_ with
+      fields of its own, the value MUST be entirely replaced, _not_ merged. This
+      means that lists of strings or lists of ints specified in a Policy overwrite
+      the empty list (in the case of a `default`) or any specified list (in the case
+      of an `override`). The same applies to `map[string]string` fields. An example
+      here would be a field that stores a map of annotations - specifying a Policy
+      that overrides annotations will mean that a final object specifying those
+      annotations will have its value _entirely replaced_ by an `override` setting.
+* In the case that two Policies of the same type specify different fields, then
+  _all_ of the specified fields MUST take effect on the affected object, using
+  the precedence rules given above.
+
+Examples to further illustrate these rules are given below.
+
+#### Attaching Policy to GatewayClass
+GatewayClass may be the trickiest resource to attach policy to. Policy
+attachment relies on the policy being defined within the same scope as the
+target. This ensures that only users with write access to a policy resource in a
+given scope will be able to modify policy at that level. Since GatewayClass is a
+cluster scoped resource, this means that any policy attached to it must also be
+cluster scoped.
+
+GatewayClass parameters provide an alternative to policy attachment that may be
+easier for some implementations to support. These parameters can similarly be
+used to set defaults and requirements for an entire GatewayClass.
+
+### Merging into existing `spec` fields
```

Comment:
This logic is fairly complex. It would be great if we could eventually offer gwctl as a library that could help with implementing this logic + ensure consistency across implementations (cc @gauravkghildiyal).